js/ui/viewSelector.js

```diff
@@ -113,6 +119,84 @@ const ShowOverviewAction = new Lang.Class({
     }
 });
 
+const ViewsDisplayLayout = new Lang.Class({
+    Name: 'ViewsDisplayLayout',
+    Extends: Clutter.BinLayout,
+
+    _init: function(appDisplayActor) {
+        this.parent();
+
+        this._appDisplayActor = appDisplayActor;
+        this._appDisplayActor.connect('style-changed', Lang.bind(this, this._onStyleChanged));
+    },
+
+    _onStyleChanged: function() {
+        this.layout_changed();
+    }
+});
+
+const ViewsDisplayContainer = new Lang.Class({
+    Name: 'ViewsDisplayContainer',
+    Extends: St.Widget,
+    Signals: {
+        'views-page-changed': { }
+    },
+
+    _init: function(appDisplay) {
+        this._appDisplay = appDisplay;
+        this._activePage = ViewsDisplayPage.APP_GRID;
+
+        this.parent({ layout_manager: new ViewsDisplayLayout(this._appDisplay.actor),
+                      x_expand: true,
+                      y_expand: true });
+
+        this.add_actor(this._appDisplay.actor);
+    },
+
+    showPage: function(page) {
```

Comment:
I still think it's cleaner to make this a property, since the code in this and the following method is exactly what you'd expect to find in a property setter and getter.